composer/lib/dependabot/composer/metadata_finder.rb

```diff
@@ -28,27 +28,26 @@ def look_up_source_from_packagist
         return nil if packagist_listing&.fetch("packages", nil) == []
         return nil unless packagist_listing&.dig("packages", dependency.name.downcase)
```

Comment:
My ruby knowledge is pretty basic so apologies if I'm missing something but are you not calling packagist_listing several times here, which would result in several requests happening to us? Even if `Dependabot::RegistryClient.get` below is caching the network bits, it still seems kinda wasteful vs just calling it once here and working with the result instead of doing 3 calls.

Reply:
`packagist_listing` is memoized in the method itself already, so this is ~the same performance wise 👍